src/Octoshift/GithubApi.cs

```diff
@@ -481,32 +481,7 @@ public virtual async Task<Mannequin> GetMannequin(string orgId, string username)
         {
             var url = $"{_apiUrl}/graphql";
 
-            var query = "query($id: ID!, $first: Int, $after: String)";
-            var gql = @"
-                node(id: $id) {
-                    ... on Organization {
-                        mannequins(first: $first, after: $after) {
-                            pageInfo {
-                                endCursor
-                                hasNextPage
-                            }
-                            nodes {
-                                login
-                                id
-                                claimant {
-                                    login
-                                    id
-                                }
-                            }
-                        }
-                    }
-                }";
-
-            var payload = new
-            {
-                query = $"{query} {{ {gql} }}",
-                variables = new { id = orgId }
-            };
+            var payload = GetMannequinsPayload(orgId);
```

Comment:
This is only partially DRY. Instead it would be better to call `GetMannequins` method here and then build or query by calling `FirstOrDefaultAsync()`. The only thing here is that `GetMannequins` should return an actual `IEnumerable` without calling `ToListAsync` so that inside `GetMannequin` method we can build our query off of the returned `IEnumerable`. Our method will look like the following after the changes:
```c#
public virtual async Task<Mannequin> GetMannequin(string orgId, string username)
{
    var url = $"{_apiUrl}/graphql";

    var payload = GetMannequinsPayload(orgId);

    var mannequin = await _client
        .GetMannequins(orgId)
        .FirstOrDefaultAsync(jToken => username.Equals((string)jToken["login"], StringComparison.OrdinalIgnoreCase));
    
    return mannequin is null ? new Mannequin() : MannequinBuilder(mannequin);
}
```

Reply:
Based on our Zoom sync up, for simplicity we decided that it would be better not to go ahead with my proposed changes for this since `GetMannequins` has to return an `IAsyncEnumerable` and it is going to make things unnecessarily complicated.